torch/_subclasses/fake_tensor.py

```diff
@@ -593,6 +593,28 @@ def nonzero(fake_mode, func, arg):
     return arg.new_empty((arg.nonzero_memo, arg.dim()), dtype=torch.int64)
 
 
+@register_op_impl(lambda func: func is torch.ops.aten.masked_select.default)
+def masked_select(fake_mode, func, self, mask):
+    if (
+        fake_mode.shape_env is None
+        or not fake_mode.shape_env.allow_dynamic_output_shape_ops
```

Comment:
I forget why we have this config (as opposed to dyn output shapes always being allowed when we have a shape env), but i'll read the code later - see we do the same for nonzero.

Reply:
The reason is inductor currently chokes on this stuff, so we need to induce a graph break under default config. But https://github.com/pytorch/pytorch/pull/109893 should make it so we can start attempting to change the defaults here.